tests/test_api.py

```diff
@@ -85,3 +91,111 @@ def test_qasm_to_phir(self) -> None:
         """
 
         assert qasm_to_phir(qasm, QtmMachine.H1_1)
+
+    def test_qasm_to_phir_with_wasm(self) -> None:
+        """Test the qasm string entrypoint works with WASM."""
+        qasm = """
+        OPENQASM 2.0;
+        include "qelib1.inc";
+
+        qreg q[2];
+        h q;
+        ZZ q[1],q[0];
+        creg cr[3];
+        creg cs[3];
+        creg co[3];
+        measure q[0]->cr[0];
+        measure q[1]->cr[1];
+
+        cs = cr;
+        co = add(cr, cs);
+        """
+
+        wasm_bytes = get_wat_as_wasm_bytes(WatFile.add)
+
+        wasm_uid = hashlib.sha256(base64.b64encode(wasm_bytes)).hexdigest()
```

Comment:
@neal-erickson I referenced https://github.com/CQCL/tket/blob/6906704b0ad64d077dff5c71454652926f91ee12/pytket/pytket/wasm/wasm.py#L77-L81 to generate this `wasm_id`